test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
What you talking looks like your personal opinion to me. And because of that I don't see why I should change my mind because your opinion equals to mine for me. It would be useful if you give me links with explanation of how we chose it should be. Also I see that integration tests create files not how you expect that they should do.